### DISCUSSION TITLE
1.5.1

### DISCUSSION THREAD

**Discussion by ilyhalight** [ilyhalight/voice-over-translation#550] [Обновления]
## Список изменений
- Добавлена поддержка Одноклассников от @SashaXser в #476
- Добавлена поддержка Google Drive (только публичные ссылки, например: `https://drive.google.com/file/d/FILE_ID`)
- Добавлена поддержка Banned Video (banned.video)
- Добавлена поддержка Egghead от @SashaXser в #531
- Добавлена поддержка Youku от @SashaXser в #531
- Добавлена поддержка корп. видео для `player.vimeo.com` (поддержка параметра app_id). Для использования перевода на `player.vimeo.com` нужно полностью отключить CSP.
- Добавлена поддержка Facebook (работает только с ссылками, которые содержат `/video/` или `/reel/`). Для использования перевода нужно полностью отключить CSP.
- Добавлена поддержка Weverse (работает только с ссылками, которые содержат `/live/` или `/media/`)
- Добавлена поддержка Newgrounds от @SashaXser в #523
- Добавлено определение языка в VK по субтитрам и обновление зависимостей от @SashaXser в #531
- Исправлена работа YouTube Embed через JS API (#544)
- Исправлена работа Bitchute от @SashaXser в #531 (#542)
- Улучшена очистка описания на Youtube от @SashaXser в #476
- Улучшена логика инициализации БД от @SashaXser в #531
- Улучшено получение ID видео для Dailymotion от @SashaXser в #476
- Улучшение оптимизации от @SashaXser в #499
- Улучшено добавление фраз локализации при сборке от @SashaXser в #499
- Функция форматирования строк заменена на стандартный `replace` от @SashaXser в #499
- Исправлен баг, когда стримы могли не переводиться, если они идут более 4 часов от @SashaXser в #499
- Исправлен баг с отключением перевода на стримах от @SashaXser в #499
- Исправлена ошибка из-за которой значение автогромкости могло иметь слишком много знаков после запятой
- Исправлена ошибка из-за которой значение автогромкости по умолчанию было 0.15%, а не 15%
- Время отображения кнопки перевода уменьшено до 1 секунды
- Убраны пути `/s/` и `/d/` из определения айди видео в Яндекс Диске (скрипт не запускался на этих путях, поэтому для конечного пользователя ничего не изменилось)
- HLS заменен на облегченную версию HLS light от @SashaXser в #506
- В утилитах для сайтов экспорт изменен на эспорт по умолчанию
- Теперь, полученные переводы кешируются до перезагрузки страницы. Благодаря этому удалось избавиться от лишних запросов к серверу, а так же это избавило от проблем, когда из-за изменения качества приходилось заново ждать перевод (например на weverse)
- Старая реализация установки громкости звука в "stopTranslate" изменена на актуальную - "this.setVideoVolume"
- Объединен некоторый повторяющийся код
- Webpack-Userscript изменен на Webpack-Monkey
- Стандартный домен worker прокси-сервера изменен на `vot.toil.cc` (Балансировщик между прокси серверами)
- Стандартный домен M3U8 Proxy Worker изменен на `m3u8-proxy.toil.cc`, дабы избавиться от ограничений cloudflare (слишком маленький лимит запросов для m3u8 + недоступность в РФ). Если вы все равно хотите использовать M3U8 Proxy Worker, который хостится на cloudflare, то прошу перейти на `m3u8-proxy.toiloff.workers.dev` т.к. воркер на старом домене будет выключен спустя некоторое время для уменьшения числа запросов к основному worker proxy.

## What's Changed
* dailymotion fix, ok.ru support + some fixed by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/476
* 1.5.1-beta2 by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/499
* Облегченный hls.js by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/506
* Удаление/комментирование неиспользуемого кода by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/508
* Более продуманная очистка by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/512
* Поддержка Newgrounds.com и мелкие правки кода by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/523
* Добавлена поддержка 2 сайтов by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/531
* 1.5.1 release by @ilyhalight in https://github.com/ilyhalight/voice-over-translation/pull/549

**Full Changelog**: https://github.com/ilyhalight/voice-over-translation/compare/1.5.0.5...1.5.1

<hr /><em>This discussion was created from the release <a href='https://github.com/ilyhalight/voice-over-translation/releases/tag/1.5.1'>1.5.1</a>.</em>

**Comment by NullVerdict** [comment]
Картинки в README перепутаны, NewGrounds не включен в список поддерживаемых сайтов.

**Reply by ilyhalight** [reply]
исправил

**Comment by astronom-v-cube** [comment]
Спасибо от души♥

**Comment by FiLinX** [comment]
ну что, можно переходить на эту версию? 
я то до сих пор сижу на 1.4.1 с её выхода)  -  за всё время ни одной проблемы!

**Reply by NullVerdict** [reply]
Попробуй

**Reply by FiLinX** [reply]
попробовал)
и вроде бы как всё работает, но увы, на 90% мной просматриваемого ютуба - не работает((
потому что я в основном смотрю ролики открывая их в embed - а вот он то и не работает(
1.5.1.1 аналогично..
вернулся пока на 1.4.1
